ceno_zkvm/src/structs.rs

```diff
@@ -518,6 +518,57 @@ impl<E: ExtensionField> ZKVMWitnesses<E> {
         } else {
             vec![]
         };
+
+        // 2. process record which init within shard-range and read by later shards.
+        let current_shard_access_later = final_mem
+            .par_iter()
+            // only process no range restriction memory record
+            // for range specified it means dynamic init across different shard
+            .filter(|(_, range, _)| range.is_some())
+            .flat_map(|(mem_name, range, final_mem)| {
+                let range = range.as_ref().unwrap();
+                final_mem.par_iter().filter_map(|mem_record| {
+                    // prepare cross shard writes record for those record which not accessed in first record
```

Comment:
typo: "not accessed in first record", should be "not accessed in current shard".